### PR TITLE
Use solc.js compiler [WIP]

### PR DESCRIPTION
**Pull request by axic**
This is a follow up of #54 and needs that PR to be merged first (has those commits here also, but will rebase).  It also depends on https://github.com/ethereum/solc-js/pull/21.

It is incomplete and work in progress. Looking for comments, @chriseth @redsquirrel

### CODE REVIEW

File: src/app/compiler-worker.js [hunk 1]
```diff
@@ -12,25 +13,22 @@ module.exports = function (self) {
...
-				{
-					compileJSONInternal = Module.cwrap("compileJSONCallback", "string", ["string", "number", "number"]);
-					var missingInputCallback = Module.Runtime.addFunction(function(path) {
-						missingInputs.push(Module.Pointer_stringify(path));
-					});
-					compileJSON = function(input, optimize) {
-						return compileJSONInternal(input, optimize, missingInputCallback);
-					};
+
+				var compiler = solc(Module);
```
Comment (redsquirrel) [inline src/app/compiler-worker.js]: Where does `Module` come from?

Comment (axic) [inline src/app/compiler-worker.js]: Loading an emscripten compiled binary in the global context will be assigned to the `Module` variable.

`importScripts(data.data);` above will load the url and put it into the global context

**Comment by chriseth** [comment]
Tests fail?

**Comment by axic** [comment]
@chriseth it needs a newer solc release of package.json needs to refer to the solc github repo

**Comment by axic** [comment]
Would be nice to find a resolution to https://github.com/ethereum/solc-js/issues/24 before merging this.

**Comment by axic** [comment]
> Would be nice to find a resolution to ethereum/solc-js#24 before merging this.

The reason for that is to avoid double JSON encoding/decoding in the case of the worker.

**Comment by axic** [comment]
@chriseth apart from the previous possible improvement, this worked last time I've checked

**Pull request merged**